INTEGRATING.md

```diff
@@ -1,34 +1,103 @@
+implement the provider side of the `loki_push_api` relation interface.
+For instance a Loki charm.
+
+- `LokiPushApiConsumer`: This object is meant to be used by any charmed operator that needs to
+send log to Loki by implementing the consumer side of the `loki_push_api` relation interface.
+For instance a Promtail or Grafana agent charm that needs to send logs to Loki.
+
+Learn more about this library [in charmhub](https://charmhub.io/loki-k8s/libraries/loki_push_api).
+
+
+#### `log_proxy`
+
+This library provides two main objects:
+
+- `LogProxyProvider`: This object can be used by any charmed operator that needs to act
+as a Log Proxy to Loki by implementing the provider side of `loki_push_api` relation interface.
+For instance a Grafana agent or Promtail charmed operator that receives logs from a workload
+and forward them to Loki.
+
+- `LogProxyConsumer`: This object can be used by any K8s charmed operator that needs to
+send log to Loki through a Log Proxy by implementing the consumer side of the `loki_push_api`
+relation interface.
+When a relation with a Charmed Operator that implements the `LogProxyProvider` is established,
+this object injects and configure into the workload container a [Promtail binary](https://grafana.com/docs/loki/latest/clients/promtail/)
+that will send logs to Loki through a Log Proxy charmed operator.
+
+
+
+Filtering logs in Loki is largely performed on the basis of labels.
+In the Juju ecosystem, Juju topology labels are used to uniquely identify the workload that
+generates telemetry like logs.
+In order to be able to control the labels on the logs pushed this object injects a Pebble layer
+that runs Promtail in the workload container, injecting Juju topology labels into the
+logs on the fly.
+
+
+Learn more about this library [in charmhub](https://charmhub.io/loki-k8s/libraries/log_proxy).
+
+
+#### Integration example.
+
+
+With both libraries `loki_push_api` and `log_proxy` the following integration can be done:
+
+
+```
+┌──────────────────┐  Workload logs  ┌────────────────────┐  Workloads logs ┌──────────────────┐
+│                  │ ───────────────►│                    │ ───────────────►│                  │
+│ Custom Workload  │                 │  Grafana agent     │                 │ Loki             │
+│                  ├─────────────────┤                    ├─────────────────┤                  │
+│ Charmed Operator │ loki_push_api   │  Charmed Operator  │  loki_push_api  │ Charmed Operator │
+│                  │                 │                    │                 │                  │
+└──────────────────┘                 └────────────────────┘                 └──────────────────┘
+Uses:                                 Uses:                                  Uses:
+- LogProxyConsumer                    - LokiPushApiConsumer                 - LokiPushApiProvider
+                                      - LogProxyProvider
+```
```

Comment:
Awesome!
Have you also considered a `.drawio.svg` or `.drawio.png` diagram format (standard format with embedded draw.io information)?

Reply:
Yes.
I thought that if we are happy with this simple diagrams we can move later to a SVG